libs/wingc/src/type_check.rs

```diff
@@ -104,76 +163,58 @@ impl PartialEq for Type {
 			(Self::Function(l0), Self::Function(r0)) => l0 == r0,
 			(Self::Class(l0), Self::Class(_)) => {
 				// If our parent is equal to `other` then treat both classes as equal (inheritance)
-				if let Some(parent) = l0.parent {
-					let parent_type: &Type = parent.into();
+				if let Some(parent) = l0.parent.as_ref() {
+					let parent_type: &Type = &*parent;
 					return parent_type.eq(other);
 				}
 				false
 			}
 			(Self::Resource(l0), Self::Resource(_)) => {
 				// If our parent is equal to `other` then treat both resources as equal (inheritance)
-				if let Some(parent) = l0.parent {
-					let parent_type: &Type = parent.into();
+				if let Some(parent) = l0.parent.as_ref() {
+					let parent_type: &Type = &*parent;
 					return parent_type.eq(other);
 				}
 				false
 			}
 			(Self::Struct(l0), Self::Struct(_)) => {
 				// If we extend from `other` then treat both structs as equal (inheritance)
 				for parent in l0.extends.iter() {
-					let parent_type: &Type = (*parent).into();
+					let parent_type: &Type = &*parent;
```

Comment:
It is strange to see `&*` everywhere knowing that it's not a no-op.

Reply:
Open for suggestions... I like it being more concise than `into()` and also not giving the feeling that I'm doing some conversion here (because I'm just casting). Might be a way to make this implicit, but for now I think it's better than what we had.

Reply:
No I agree that this way is better, no suggestions from me until we revisit the safety of the whole thing